### PR TITLE
Derive debug with array too big

### PR DESCRIPTION
**Pull request by fitzgen**
See the first commit's message for details, second commit is just `rustfmt`.

r? @emilio 

Not quite sure whether the `Opaque` struct is fully motivated... it seems like it would be useful later on in codegen, but might not be pulling its own weight without more usage...

### CODE REVIEW

**Comment by fitzgen** [comment]
Overall, I think that adding the `CanDerive{Copy,Debug}` traits is a nice clean up, though :)

File: libbindgen/src/ir/layout.rs [hunk 1]
```diff
@@ -31,4 +35,32 @@ impl Layout {
...
+}
+
+/// When we are treating a type as opaque, it is just a blob with a `Layout`.
+pub struct Opaque(pub Layout);
+
+impl CanDeriveDebug for Opaque {
+    type Extra = ();
+
+    fn can_derive_debug(&self, _: &BindgenContext, _: ()) -> bool {
+        self.0.size < RUST_DERIVE_IN_ARRAY_LIMIT
```
Comment (emilio) [inline libbindgen/src/ir/layout.rs]: This is not right (or at least, not as permissive as it should). Size is in bytes, but we sometimes generate arrays of `u32`, `u64` to preserve alignment.

This code should check that to decide it too.

**Comment by emilio** [comment]
Looks great!

@bors-servo r+

**Comment by fitzgen** [comment]
Thanks :)

**Pull request merged**